src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -220,7 +228,48 @@ void AppHost::_HandleCommandlineArgs()
         // is created.
         if (_windowManager.IsMonarch())
         {
-            _logic.SetNumberOfOpenWindows(_windowManager.GetNumberOfPeasants());
+            const auto numPeasants = _windowManager.GetNumberOfPeasants();
+            const auto layouts = ApplicationState::SharedInstance().PersistedWindowLayouts();
+            if (_logic.ShouldUsePersistedLayout() && layouts && layouts.Size() > 0)
+            {
+                uint32_t startIdx = 0;
+                // If we are the only window, and no commandline arguments were provided
+                // then we should just use the current window to load the first layout
+                if (numPeasants == 1 && !_logic.HasCommandlineArguments())
+                {
+                    _logic.SetPersistedLayoutIdx(startIdx);
+                    startIdx += 1;
+                }
+
+                // Create new windows for each of the other saved layouts.
+                for (auto size = layouts.Size(); startIdx < size; startIdx += 1)
+                {
+                    auto newWindowArgs = fmt::format(L"{0} -w new -s {1}", args[0], startIdx);
+
+                    STARTUPINFO si;

```

Comment:
You should probably use a [`wil::unique_process_information`](https://github.com/microsoft/wil/wiki/RAII-resource-wrappers#wilunique_struct) here

Reply:
I was waiting for someone to tell me how to do this better than what I copied off of MSDN.